app/src/main/java/in/ac/nitc/eyyauto/MainActivity.java

```diff
@@ -2,70 +2,69 @@
 
 import android.content.Intent;
 import android.os.Bundle;
-import android.support.annotation.NonNull;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
+import android.widget.EditText;
 import android.widget.Toast;
 
 import com.firebase.ui.auth.AuthUI;
 import com.firebase.ui.auth.ErrorCodes;
 import com.firebase.ui.auth.IdpResponse;
-import com.google.android.gms.tasks.OnCompleteListener;
-import com.google.android.gms.tasks.Task;
 import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DatabaseError;
 
 import java.util.Arrays;
 
+import in.ac.nitc.eyyauto.handlers.Event;
+import in.ac.nitc.eyyauto.handlers.UserHandler;
+import in.ac.nitc.eyyauto.models.User;
+
 public class MainActivity extends AppCompatActivity {
 
     private static int RC_SIGN_IN = 123;
+    private FirebaseUser mUser;
+    private UserHandler mUserHandler;
+
+    private Button mConfirm;
+    private EditText mNameField;
+    private String mUserId;
+    private String mName;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
-        Button signInButton = findViewById(R.id.button);
-        signInButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                SignIn();
-            }
-        });
-    }
-
-    private void SignIn() {
-        FirebaseAuth mAuth = FirebaseAuth.getInstance();
-        if (mAuth.getCurrentUser() != null) {
-            Toast.makeText(this, "Signing " + mAuth.getCurrentUser().getDisplayName() + " out...", Toast.LENGTH_SHORT).show();
-            AuthUI.getInstance()
-                    .signOut(this)
-                    .addOnCompleteListener(new OnCompleteListener<Void>() {
-                        @Override
-                        public void onComplete(@NonNull Task<Void> task) {
-                            Toast.makeText(MainActivity.this, "Signed out", Toast.LENGTH_SHORT).show();
-                        }
-                    });
+        // set view according to need
+        Bundle extras = getIntent().getExtras();
+        if(extras == null) {
```

Comment:
Instead of this, create class variables that store the content of the extras, which can then be used to configure how this activity runs.
```suggestion
        private boolean hasPhoneNumber;
        .
        .
        hasPhoneNumber = extras.get("hasPhoneNumber");
```
Then in onResume() we have:
```suggestion
        if(hasPhoneNumber){
        // do something
        }
```